chapters/en/slides/module4/module4_01.Rmd

```diff
@@ -0,0 +1,345 @@
+
+```{python}
+alt.Chart(movies_extended).mark_point().encode(
+    alt.X('Production Budget'),
+    alt.Y('Worldwide Gross'))
+```
+
+Notes:
+Since we are interested in comparing two columns,
+we will add the second column to the y-axis.
+
+Sometimes this is enough to avoid overplotting
+since we are now spreading the points across two dimensions
+instead of one.
+
+However,
+as you can see in the plot on this slide,
+it is still impossible to tell
+if there are more points close to 80 million of 0,
+and likewise 200 million or 0 on the y-axis.
+
+So although we can discern a trend
+for the points outside the saturated area
+we do not know how they data is spread out inside the blue blob. 
+
+We saw previously that 
+histograms can help when we have saturated plots.
+But if we were to make a histogram
+of each column separately
+we would have two one-dimensional histogram 
+but no information of how these relate to eachother.
+
+Instead,
+what we need here is a single two-dimensional histogram
+of the scatter plot above.
+
+---
+
+## A heatmap allows us to visualize the relationship between two distributions
+
+```{python}
+alt.Chart(movies_extended).mark_rect().encode(
+    alt.X('Production Budget', bin=alt.Bin(maxbins=60)),
+    alt.Y('Worldwide Gross', bin=alt.Bin(maxbins=60)),
+    alt.Tooltip('Worldwide Gross', bin=alt.Bin(maxbins=60)),
+    alt.Color('count()'))
+```
+
+Notes:
+What's involved in creating a two-dimensional histogram?
+First, we need to bin both axes instead of just one.
+The bins will look like a grid or mesh
+overlayed on the image,
+similar to the faint grey gridlines in the previous slide.
+
+Wihtin each rectangle of this grid,
+we will count the number of observations
+and represent the count value with a color.
+
+The result of these operations is the heatmap shown in this slide,
+which enables us to see a level of detail we could not perceive in the scatter plot.
+Here it is clear that there are many fewer movies with a production budget of 80 million
+comared to the area close to 0,
+and most movies seem to be around 10-15 million.
+Likewise the grossing of most movies is around 0-50 million,
+not 200 million.
+
+---
+
+## A density plot can also be extended to two dimensions
+
+<img src="/module4/unnamed-chunk-5-1.png" width="48%" />
+
+```{python}
+import seaborn as sns
+
+sns.kdeplot(
+    data=movies_extended,
+    x='Production Budget',
+    y='Worldwide Gross',
+    fill=True, cmap='YlGn', cut=0)
+sns.despine()
+```
+
+Notes:
+In addition to making 2D histograms,
+we can also make 2D density plots.
+Each kernel is now two dimensionalal
+and look like a bell or the top of a circus tent,
+rather than a bell-shaped one dimensional curve.
+
+Altair cannot yet make thes plots,
+so here we're showing an example from another plotting library called seaborn,
+so that you can get a sense of what this visualization would look like for our data.
+
+TODO testing to involve seaborn as per our discussion today.
+
+---
```

Comment:
I think summary slides are a good idea in general, but I have to cut some things for time and since we don't have summary slides anywhere else, let's not include them here. I will open an issue for discussing having summary slides somewhere, but I also think the interactive questions we have serve some of the same purpose in reminding students about what they just learned.